database/migration/databaseMethods.ts

```diff
@@ -479,6 +480,103 @@ const databaseMethods = {
       })
     }
   },
+  secureDataTables: async () => {
+    const dataTables = (
+      await DBConnect.query({ text: `SELECT table_name FROM data_table;`, rowMode: 'array' })
+    ).rows.flat()
+
+    for (const table of dataTables) {
+      try {
+        await DBConnect.query({
+          text: `ALTER TABLE ${config.dataTablePrefix}${table} ENABLE ROW LEVEL SECURITY;`,
+        })
+      } catch {
+        console.log(`ERROR: Problem securing data table "${table}"`)
+      }
+    }
+  },
+  addDataViewsForLookupTables: async () => {
+    try {
+      await DBConnect.query({ text: `SELECT menu_name FROM data_view` })
+      console.log('  ...already done, skipping...')
+      // If this column exists, then we've done this before, so don't re-run the
+      // migration in the `catch` block (otherwise it'll add menu names to
+      // existing lookup table views)
+    } catch {
+      // Add new column to data views
+      await DBConnect.query({
+        text: `ALTER TABLE data_view
+        ADD COLUMN IF NOT EXISTS menu_name varchar;`,
+      })
+
+      // Fill this column with default "title" value for existing
+      await DBConnect.query({
+        text: `UPDATE data_view
+          SET menu_name = title;`,
+      })
+    }
+    // Add dataView codes for lookup tables that don't have one
+    const lookupTablesWithoutDataView = (
+      await DBConnect.query({
+        text: `
+      SELECT id FROM data_table
+        WHERE data_view_code IS NULL
+        AND is_lookup_table = true;`,
+        rowMode: 'array',
+      })
+    ).rows.flat()
+    const randomId = customAlphabet('1234567890', 4)
+    for (const id of lookupTablesWithoutDataView) {
+      await DBConnect.query({
+        text: `UPDATE data_table
+          SET data_view_code = CONCAT(LEFT(table_name, 3), ${randomId()})
```

Comment:
Lookup tables get a random-ish code for their default data view (e.g. "product" -> "pro7364"). This is never seen in UI by default, and can be changed to something "nicer" if ever the lookup table needs to be accessed via "Database" (Data view) menu.